syntax = "proto3";

package pb;

enum ColorBlindnessType {
    PROTANOPIA = 0;
    DEUTERANOPIA = 1;
    TRITANOPIA= 2;
}

enum TaskType {
    NOTHING = 0;
    EPILEPSY = 1;
    COLOR_BLINDNESS = 2;
    EPILEPSY_AND_COLOR_BLINDNESS = 3;
}

message SegmentRequest {
    string filename = 1;
    int64 number = 2;
    TaskType taskType = 3;
    optional ColorBlindnessType colorBlindnessType = 4;
}

message SegmentResponse {
    string filename = 2;
    bytes segment = 1;
}

message VideoMetadataRequest {
    string filename = 1;
}

message VideoMetadataResponse {
    string filename = 2;
    bytes videoMetadata = 1;
}
